src/components/HydrantList.js

```diff
@@ -27,49 +27,44 @@ class HydrantList extends React.Component {
 
   state = {
     open: true,
-    showDetails:{}
+    showDetails: {}
   }
 
-render() {
-
-  const {
-    classes,
-    editableTrail,
-    hydrants,
-    hydrantDeleted,
-    modifyHydrant
-  } = this.props;
-
-  const trailHydrants = hydrants
-      .filter((h) => h.get('trail') === editableTrail.get('id'))
-      .sort((a,b) => a.get('name').localeCompare(b.get('name'), undefined, {numeric: true}))
-      .valueSeq()
-
-  const toggleHighLight = (feature, state) => {
+  toggleHighLight = (feature, state) => {
     feature.set('highlighted', state)
     feature.changed();
   }
 
-  const updateCoords = (e, h, coordIndex) => {
-    const newCoords = _.clone(h.get('coords'));
-    newCoords.splice(coordIndex, 1, Number(e.target.value))
-    modifyHydrant(h.get('id'), { coords: newCoords })
-  };
-
-  const toggleEdit = (id) => {
+  toggleEdit = (id) => {
     const newDetails = {}
-    if (!this.state.showDetails[id]){
+    if (!this.state.showDetails[id]) {
       newDetails[id] = true
```

Comment:
This isn't actually a toggle since it can only open things and not close them. If you to be able to re-collapse some hydrant forms you should put that in here.

Reply:
This actually should both open and close the editForm for a given hydrant since it only sets the States NewDetails if the passed id is different from the states. Otherwise it sets it back to an empty object and all are collapsed since the Collapse in value is based on newDetails. Now that we've seperated HydrantItems though, I can make this simpler.